pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -811,26 +806,22 @@ func (p *PriorityQueue) done(pod types.UID) {
 	// remove events which is only referred from this Pod
 	// so that the receivedEvents map doesn't grow infinitely.
 	event := p.receivedEvents.Front()
-	if inFlightPod.previousEvent != nil {
+	if inFlightPod.previousEvent != nil && inFlightPod.previousEvent.Next() != nil {
 		event = inFlightPod.previousEvent.Next()
 	}
-	for ; event != nil; event = event.Next() {
+	for event != nil {
```

Comment:
Is it change intentional to catch some bug? If not, the previous version was more readable IMO.

Reply:
If we kept `; event != nil; event = event.Next()`, `e.inFlightPodsNum <= 0` case would be complicated since event.Next() doesn't specify the next event after the event is removed from the list.